### PR TITLE
few things for compatability with latest pandas

### PR DESCRIPTION
**Pull request by andrewgsavage**
A couple of imports from pandas have moved around/don't exist any more. This fixes them.

### CODE REVIEW

File: pintpandas/pint_array.py [hunk 1]
```diff
@@ -81,7 +82,7 @@ def __new__(cls, units=None):
 
     @classmethod
     def _parse_dtype_strict(cls, units):
-        if isinstance(units, compat.string_types):
```
Comment (jbrockmendel) [inline pintpandas/pint_array.py]: if you want to maintain py2 compatibility, six.string_types is identical to the now-gone pandas.compat.string_types.

**Comment by hgrecco** [comment]
Shall we also change the minimum required pandas version?

**Comment by jbrockmendel** [comment]
ExtensionArray should work fine in 0.24.0 and on

File: pintpandas/pint_array.py (hunk 1 above)
Comment (hgrecco) [reply pintpandas/pint_array.py]: I think it would be better to use pint compat module which also contains a `string_types`

**Comment by hgrecco** [comment]
> ExtensionArray should work fine in 0.24.0 and on

Awesome. Let's just remove the `rc1` then just to make things cleaner.

**Comment by andrewgsavage** [comment]
removed the rc1

**Comment by andrewgsavage** [comment]
This fixes imports that moved around for pd 0.25.

There's some tests that need setting up fixtures/customising for pint. I think they should be done in a later PR.

File: pintpandas/pint_array.py [hunk 2]
```diff
@@ -300,7 +301,7 @@ def astype(self, dtype, copy=True):
         array : ndarray
             NumPy ndarray with 'dtype' for its dtype.
         """
-        if isinstance(dtype,str) and (dtype.startswith("Pint[") or dtype.startswith("pint[")):
+        if isinstance(dtype,string_types) and (dtype.startswith("Pint[") or dtype.startswith("pint[")):
```
Comment (jbrockmendel) [inline pintpandas/pint_array.py]: does pint have a policy about linting/style nitpicks?  if so, a space after the comma here would be good.

**Comment by Nikolai-Hlubek** [comment]
from pint.compat import string_types
doesn't work with latest pint.

**Comment by hgrecco** [comment]
I will merge this and then fix the other things

**Comment by hgrecco** [comment]
fixed in master

**Pull request closed without merging**